phpseclib/File/ASN1.php

```diff
@@ -590,7 +590,7 @@ function asn1map($decoded, $mapping, $special = array())
                                 $maymatch = $constant == $temp['constant'] && $childClass == $tempClass;
                             } else {
                                 // Can only match if no constant expected and type matches or is generic.
-                                $maymatch = !isset($child['constant']) && array_search($child['type'], array($temp['type'], self::TYPE_ANY, self::TYPE_CHOICE)) !== false;
+                                $maymatch = !isset($child['constant']) && in_array($child['type'], array($temp['type'], self::TYPE_ANY, self::TYPE_CHOICE));
```

Comment:
I think there is a special case where array_search vs in_array matters, but can't remember right now. Not sure it makes a difference in this case. Probably not.